go-selinux/label/label_selinux.go

```diff
@@ -118,7 +118,13 @@ func SocketLabel() (string, error) {
 // SetKeyLabel takes a process label and tells the kernel to assign the
 // label to the next kernel keyring that gets created
 func SetKeyLabel(processLabel string) error {
-	return selinux.SetKeyLabel(processLabel)
+	err := selinux.SetKeyLabel(processLabel)
+	if err != nil && processLabel == "" {
+		if os.IsNotExist(err) || os.IsPermission(err) {
```

Comment:
Why would we block permission denied?

Reply:
Because in this case, the path `/proc/self/attr/keycreate` is exist, when we write to it:
(1) with empty string, it got an err: `container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"write /proc/self/attr/keycreate: permission denied\""`;
(2) with non empty string, it write success.

Reply:
What platform is this on?  I would say this is correct behaviour.
What kernel are you running and what is the protections on /proc/self/attr/keycreate?

Reply:
If you are running with SELinux disabled this should also not cause a write.

Reply:
Never mind, now I understand what you are doing.

Reply:
> Never mind, now I understand what you are doing.

No, I think you may not understand what's my doing fully.
I'll write it more clearly.

Reply:
Don't worry, I will dig into it to submit a more reasonable commit.
If we are hurry to fix it in runc, I think https://github.com/opencontainers/runc/pull/2033 is safe enough.
If we are don't hurry, please make me work success on this.

Reply:
I have remove `os.IsPermission`. Please see https://github.com/opencontainers/selinux/pull/52#issuecomment-480142795.